core/src/banking_stage/transaction_scheduler/transaction_state_container.rs

```diff
@@ -0,0 +1,311 @@
+use {
+    super::{
+        transaction_priority_id::TransactionPriorityId,
+        transaction_state::{SanitizedTransactionTTL, TransactionState},
+    },
+    crate::banking_stage::scheduler_messages::TransactionId,
+    min_max_heap::MinMaxHeap,
+    solana_runtime::transaction_priority_details::TransactionPriorityDetails,
+    std::collections::HashMap,
+};
+
+/// This structure will hold `TransactionState` for the entirety of a
+/// transaction's lifetime in the scheduler and BankingStage as a whole.
+///
+/// Transaction Lifetime:
```

Comment:
Wdyt about the lifetime in previous diagram?

Reply:
What I've got implemented right now is more like:
```mermaid
graph TB;

Sigverify(((Sigverify)))
Dropped(((Dropped)))
Accepted
Forwarded

Received
    subgraph Container
    Ready["Ready/Unprocessed"]
    Pending["Processing/Pending"]
end

Sigverify --> Received
Received -->|failed validation| Dropped
Received -->|validated| Accepted
Accepted --> Ready
Ready -.->|dispatched for forwarding| Forwarded
Ready -->|dispatched for processing| Pending
Pending -->|unsuccessful process| Ready
Pending -->|successful process| Processed
Processed --> Dropped
Forwarded -->|without hold| Dropped
```

The dotted line for forwarding is because we only move out of the Ready/Unprocessed state if the decision is `Forward`. They will remain in `Ready/Unprocessed`, with the forwarded field now set to true, if the decision is `ForwardAndHold`.

Potentially a more ideal path is:

```mermaid
graph TB;

Sigverify(((Sigverify)))
Dropped(((Dropped)))
Accepted
Duplicate((.))
Forwarded

Received
    subgraph Container
    Ready["Ready/Unprocessed"]
    Pending["Processing/Pending"]
end

Sigverify --> Received
Received -->|failed validation| Dropped
Received -->|validated| Accepted

Accepted -->|Forward| Forwarded
Accepted -->|ForwardAndHold| Duplicate
Accepted --> Ready
Duplicate --> Forwarded & Ready
Ready -->|dispatched for forwarding| Forwarded
Ready -->|dispatched for processing| Pending
Pending -->|unsuccessful process| Ready
Pending -->|successful process| Processed
Processed --> Dropped
Forwarded --> Dropped
```

In this approach we either forward packets immediately after receiving them, or after we drain the entire container to forward.

Reply:
The second design would work better imo. But can do it incrementally.